%!TEX program = pdflatex
\documentclass{llncs}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
% \usepackage{natbib}
\usepackage[version=4]{mhchem}
\usepackage{graphicx}
\usepackage{todonotes}
% \usepackage{subcaption}
% \usepackage{wrapfig}
% \usepackage{subfig}


\begin{document}

\pagestyle{plain}
\title{Estimation of Reaction Constants Triggered by Electron Transfer in Top-Down Mass Spectrometry}
\author{Michał Aleksander Ciach\inst{1} \and Mateusz Krzysztof Łącki\inst{1} \and Błażej Miasojedow\inst{1} \and Frederik Lermyte\inst{2,3} \and Dirk Valkenborg\inst{3,4} \and Frank Sobott\inst{2} \and Anna Gambin\inst{1} }

% \author{Michał Ciach\inst{1} \and Mateusz Krzysztof Łącki\inst{1} \and Błażej Miasojedow\inst{1} \and Frederik Lermyte\inst{2,3} \and Dirk Valkenborg\inst{3,4} \and Frank Sobott\inst{2,5,6} \and Anna Gambin\inst{1} }

\institute{
Faculty of Mathematics, Informatics and Mechanics, University of Warsaw, Poland, \and
Biomolecular and Analytical Mass Spectrometry Group, Dept. of Chemistry, University of Antwerp, Belgium, \and
Centre for Proteomics, University of Antwerp, Belgium,\and
Interuniversity Institute for Biostatistics and Statistical Bioinformatics, Hasselt University, Hasselt, Belgium
% \and
% Astbury Centre for Structural Molecular Biology, University of Leeds, Leeds LS2 9JT, United Kingdom, \and
% School of Molecular and Cellular Biology, University of Leeds, LS2 9JT, United Kingdom.
}

\maketitle
\begin{abstract}
Electron transfer dissociation (ETD) is a versatile technique used in mass spectrometry for high-throughput characterization of proteins. It consists of several competing reactions triggered by the transfer of an electron from its anion source onto the sample cations. One can retrieve relative quantities of their products from mass spectra.

We present a method to analyse these results from the perspective of the reaction kinetics. A formal mathematical model of the ETD is introduced and parametrized by intensities of the existing reactions. Also, we introduce a method to estimate the reaction intensities by solving a nonlinear optimization problem. The presented method proves highly robust to noise on in silico generated data. Moreover, the presented model can explain a considerable amount of experimental results obtained under various experimental settings.
\end{abstract}

\section{Introduction}

% Owing to the important role proteins play in various biological processes, as well as the significantly greater complexity of the proteome compared to the genome \cite{Smith2013-km} there is significant need for analytical techniques capable of high-throughput protein characterization. Mass spectrometry (MS) is a highly valuable technology in this context; however, automated methods for (tandem) MS data analysis are somewhat lagging, particularly for top-down (i.e. without enzymatic digestion of proteins before MS analysis) and native (i.e. while preserving noncovalent interactions during sample preparation and in the gas phase) approaches.

\textbf{Motivation.} One of the principal fragmentation methods used in top-down mass spectrometry is Electron Transfer Dissociation, ETD, which is based on the interaction of a multiply charged, non-radical protein/peptide cation and a radical reagent anion \cite{Syka2004-rg,Zhurov2013-ua}. However, while this method is becoming ever more ubiquitous in MS-based proteomics analyses, important questions remain regarding the precise reaction mechanism, and which level(s) of protein structure can be probed using ETD \cite{Sohn2009-zv,Sohn2015-rp}. Shedding more light on the nature of ETD can therefore lead to optimization of instrumental settings and to better identification of peptide sequences and post-translational modifications.

There are several other fragmentation techniques, most importantly the Collision Induced Dissociation, CID, where the cleavage is induced by collisional activation, i.e. bombardment with non-reactive molecules \cite{Mitchell_Wells2005-gn}. A major disadvantage of the CID is that it often leads to loss of posttranslational modifications, particularly phosphorylation \cite{Kim2012-yz}. The Electron Transfer Dissociation has been also found to provide more uniform fragmentation than the CID, which breaks mostly the weakest bonds \cite{Kim2012-yz,Zhurov2013-ua}. However, a notable amount of work has been devoted to analyzing and mathematical modelling of the CID reaction; see, for example, \cite{Zhang2004-fp,Zhang2005-jn,Wysocki2000-am}, while the ETD has received less attention.

The fragmentation in ETD is induced by the transfer of an electron from a radical anion to the sample peptide/protein cation and results in a cleavage of one of the peptide bonds \ce{N-C_\alpha} in the protein’s backbone. The sample cations get positively charged during the electrospray ionization (ESI) step \cite{Fenn1989-mp} when protons are attached to the sample molecules, leading to the formation of \ce{[M + nH]^{n+}} ions, i.e. adding both charge and mass to the analyte molecule M.

Anions and cations may interact in several ways (see Fig.~\ref{img::reactions}):
\begin{enumerate}
        \item in the \textit{default} reaction, ETD, the cation accepts the electron and, after rapid neutralization of charge and a series of electron rearrangements, a backbone \ce{N-C_\alpha} bond breaks leading to the formation of so called \ce{c} and \ce{z} fragment where c-ion contains the N-terminus and z-ion the C-terminus.

        \item the cation accepts the electron but no fragmentation is observed, and the electron stays on the analyte. As no dissociation occurs, it is called ETnoD.

        \item one of the cation’s protons is transferred to the anion; a situation referred to as a proton transfer reaction, or PTR.
\end{enumerate}
\begin{figure}[h]\centering
        \includegraphics[width=.8\textwidth]{reactions.png}
        \caption{Studied chemical reactions  and
        idealization of the deconvolution of isotopic structures performed by {\sc MassTodon} (right).}\label{img::reactions}
\end{figure}



The appearance of the ETnoD fragments in the experimental data can be linked to the folding of proteins: although backbone cleavage occurs, noncovalent interactions keep the resulting fragments from separating, see \cite{Lermyte2014-vu,Lermyte2015-oy}. The ETnoD can also be caused by accommodation of an electron on the backbone of the protein, e.g. in an aromatic side chain. It is assumed that, regardless of the precise reaction mechanism, the electron obtained by ETnoD causes neutralization of an ESI-generated proton \cite{Lermyte2015-lm}, referred to as the \textit{quenched proton} further on.

In all of the reactions described above, the charge of a protein cation is reduced by one. The mass of electrons can be neglected, falling beyond the resolving power of most modern instruments. The isotope distributions of PTR and ETnoD products will show considerable overlap, especially for large molecules, as illustrated in Fig.~\ref{fig::massTodon}. Cations can undergo several reaction events, being multiply approached by different anions. However, the so-called internal fragments of proteins, i.e. resulting from two backbone cleavage events, are usually not observed suggesting that double ETD scarcely ever occurs. On the other hand, there is a lot of evidence of multiple ETnoD and PTR occurring on one analyte molecule  \cite{Lermyte2015-li}. Note that only charged ions are observed in the mass spectrometer.
\begin{figure}[h]\centering
        \includegraphics[width=.8\textwidth]{masstodon.png}
        \caption{The deconvolution of isotopic structures performed by {\sc MassTodon}: the mixture of two distributions is represented by a convex combination of two theoretical isotopic patterns.}\label{fig::massTodon}
\end{figure}


The peptide bond cleavage induced by ETD is believed to be fairly uniform. A notable exception from this rule is the peptide bond of proline: due to the ring structure of this amino acid, the \ce{c}- and \ce{z}-ions are held together even after the \ce{N-C\alpha} bond cleavage. A specific type of \ce{N-C\alpha} bond cleavage occurs on the N-terminus, leading to a loss of one ammonia molecule. The precise mechanism of this reaction is not known. However, the current work we assume this reaction to be an ETD, and the ammonia molecule is treated as a \ce{c} fragment. Therefore, the number of possible ETD cleavage sites is equal to the number of amino acids other than proline in the protein/peptide sequence.

\textbf{Related research.}
Various approaches have been taken to model electron transfer driven reactions \cite{Breuker2004-az,Simons2010-gy,Zhurov2013-ua,Turecek2013-fq}. A somewhat similar approach to the one taken by us was presented by Zhang \cite{Zhang2004-fp,Zhang2005-jn,Zhang2011-lg}, who uses a kinetic model to study fragmentation.
In \cite{Zhang2010-fp}, Zhang adapts the model to explicitly model mass spectra obtained with the use of ETD. The model relies upon 280 parameters and its derivation is grounded in the theory of statistical mechanics. The model was fitted to a training data set consisting of more than 7000 ETD spectra simultaneously.

On the other end of the possible spectrum of approaches to model protein fragmentation, a notable amount of literature has been built up around the idea of purely data-driven prediction of the intensity of peptides in tandem MS experiments \cite{Elias2004-fr,Arnold2006-wn,Degroeve2013-ej} . A more exploratory approach targeted at studying fragmentation patterns was taken by Li \cite{Li2011-mq}. That said, the above approaches have been applied mainly to study the Collision Induced Dissociation.

\textbf{Our contribution.}
We propose a formal model of the electron driven reactions occurring inside the mass spectrometer. We follow a modelling strategy first developed by Gambin \cite{Gambin2010} to study the degradation of proteins induced by various peptidases.
% A particular feature of this phenomenon is the irreversibility of the process: peptides cannot merge back into bigger compounds. This simple trait leads to a considerable simplifications in the dynamics of the continuous time Markov Jump Process used to model the phenomenon: the average numbers of proteins at a given time can be obtained by solving a system of recursively dependent ordinary differential equations, ODEs.
% Likewise, in the context of ETD, during each reaction the cation loses at least one charge and it is impossible for it to charge itself back; thus, there are no cycles present in the dynamics of this problem.
The solution to the problem of ETD reaction can be obtained conceptually in the same way: the stochastic description based on Markov Jump Process, MJP, can be transformed to a populational description of a large number of molecules based on Ordinary Differential Equations, ODEs.
Given initial values of parameters describing the intensities of transitions in the process, one can solve the ODEs with a recursive algorithm. The parameters can be optimized to minimize the difference between the calculated averages and the actual experimental data.

We study mass spectra gathered in controlled experiments, obtained for highly purified compounds. The identity of the precursor ion and all fragments obtained given a set of possible reactions is known and the quantities of these fragments can be established using our in-house developed identification tool called {\sc MassTodon} \cite{Lermyte2015-lm,Lermyte2017-zt}. Given a mass spectrum, {\sc MassTodon} outputs a list of reaction products together with their estimated intensities (that are usually assumed to be proportional to the actual number of ions). {\sc MassTodon} merges peaks that can be traced to originate from different isotopologues of the same molecule and also deconvolves isotope clusters into their sources or origin, see Fig.~\ref{fig::massTodon}.
The obtained information is more compact, but still not easy to interpret: the number of possible fragments is linear in the length of the amino acid sequence, excluding internal fragments that cannot be observed in experiments, and quadratic in the number of charge.

Here, we propose a method that can reduce the dimensionality of the data obtained by {\sc MassTodon}. It also allows us to interpret the spectra in terms of reaction intensities: the key notion in the theory of reaction kinetics. Albeit, it is natural to assume that the resulting spectra are produced as an outcome of some stochastic chemical process governed by the intensities of possible reactions.

The model we propose has a number of parameters linear in the length of the studied protein, reducing the quadratic term related to charge in {\sc MassTodon}. However, all the parameters are probabilities and can be aggregated at will, which leads to a massive reduction in their number. This lets us express the mass spectrum in terms of parameters such as the total intensity of reactions and the probabilities of the three studied reaction: ETD, PTR and ETnoD. A process described by a handful of parameters can be easily visualized and thus -- easily understood. Also, comparison between different spectra, e.g. coming from different instrumental settings, is highly simplified.

\textbf{Organization of the paper.}
First, we introduce the theoretical considerations behind our model. Then, we describe the procedures used to obtain our data sets (experimental and \textit{in silico}). Then, we assess the performance of the model. Finally discuss existing problems and possible extensions.


\section{Formal model of the ETD reaction}

Following the ideas outlined in \cite{Gambin2010}, we model the ETD as a continuous time Markov Jump Process, MJP. To describe the state space for our model we introduce the \textit{reaction graph}: a bipartite directed graph with two types of nodes we call \textit{molecular species} and \textit{reactions}. The molecular species correspond to cations that are substrates or products of the studied reactions, see Fig.~\ref{img::petrinet}. Each molecular species $u$ can be uniquely described by: (1) the sequence of amino acids $s$, (2) the charge of the cation $q$, and (3) the number of protons $g$ that have been paired with electrons, $u = (s,q,g)$. Storing $g$ is necessary to correctly calculate the mass of cation. All molecules that cannot be observed, e.g. analyte molecules that lost all charged protons or inner fragments, are merged into one molecular species called the \textit{cemetery}. We assume to know only the numbers of protons and not their positioning. We also assume that ESI-generated protons can only be attached to basic amino acids.
\begin{figure}[h]
        \center
        \includegraphics[width=\textwidth]{petrinet.png}
        \caption{A simplified Markov Jump Process of the ETD process. The \textit{molecular species} are depicted in black and the \textit{reactions} in orange. The dagger symbolizes the \textit{cemetery} -- a species that collects all molecules introduced to the instrument that lost their charge during in the cascade of reactions. The reaction graph serves as a board for \textit{tokens}. Counts of tokens are plotted in red in panel (B). During each reaction a token disappears on the substrate side and product tokens appear: one in case of ETnoD and PTR, two in case of ETD, as seen in (B).}\label{img::petrinet}
\end{figure}

The molecular species are occupied by tokens that represent the numbers of given molecular species. Denote the number of tokens at place $u$ by $x_u$. The state $x$ of the MJP is defined as a collection of all such counts at a given moment in time, so that $x = (x_u)$.  From a state $x$ the system can evolve to another state following one of possible reactions, check Fig.~\ref{img::petrinet} (B).

We investigate $L$ fragmenting reactions, where $L$ equals to the number of amino acids of the protein under study minus the number of prolines that ETD cannot fragment. We also consider two other reactions corresponding to ETnoD and PTR, that change one substrate into one product.

% In the continuous time MJP moments of reactions are stochastic
% reactions are drawn from a precise distribution. Another distribution describes when a reaction can happen. Here, time is
We model the time of experiment as an interval $[0,1]$, where $0$ is the beginning of the process and $1$ corresponds to the moment of observation. The probability that $X$ ends up in state $x$ at a given time $t$ is denoted by $p_x^t$. The derivative of this quantity follows the master equation,
        $$\dot{p}_x^t = \sum_{y\not=x} p_y^t Q_{yx} - p_x^t \sum_{y\not=x} Q_{xy},$$
Above, $Q_{xy}$ is the intensity of the jump from state $x$ to $y$. The intensity equals $0$ if $y$ cannot be obtained from $x$ with reaction $R$. Otherwise, the intensity is proportional to the number of the substrate molecules, $Q_{xy}=c_R x_{s_R}$,  where $c_R$ is the intensity constant described later on.

The average numbers of cations at that place $u$ is $E_u^t= \sum_x x_up_x^t$.
At $t=0$, the process is deterministic: all tokens can be found only in the precursor node with maximal charge state -- the root, $r$.
Thus, $E_r^0=N$ and $E_u^0=0$ if $u \not= r$.
These averages evolve according to
        $$\dot E_u^t=\sum_x x_u \sum_{y\not=x} p_y^t Q_{yx} -\sum_x x_u p_x^t \sum_{y\not=x} Q_{xy}.$$
The double sums iterate over tuples $(x,y)$ or $(y,x)$ corresponding to particular changes in the placement of tokens.
Unreachable states $y$ have zero intensities, $Q_{xy} = 0$.
It follows that
        $$\dot E_u^t = \sum_R cR \sum_{x:x=Ry}  x_u (x_{s_R}+1) p_{R^{-1}x}^t- \sum_R cR \sum_{x} x_u x_{s_R} p_x^t,$$
where $Ry$ denotes the state obtained from $y$ after reaction $R$ and $R^{-1}x$ denotes the state that results in $x$ when $R$ happened. For a given $R$,
        $$\sum_{x:x=Ry} x_u (x_{s_R}+1) p_{R^{-1}x}^t = \sum_{y}  (Ry)_u y_{s_R} p_y^t = \sum_x (Rx)_u x_{s_R} p_x^t,$$
as we can restate the sums in terms of states from before transition $R$ and retag $y$ for $x$.
Thus, $\dot E_u^t = \sum_R c_R \sum_x   \left[(Rx)_u-x_u \right] x_{s_R} p_x^t$. The term $\left[(Rx)_u-x_u\right]$ equals $1$ if place $u$ is a product of reaction $R$, $-1$ for substrate, and $0$ otherwise.
Therefore, it is a value dependent on $R$ and not on particular state $x$.
Denote it by $K_R$.
Then
        $$\dot E_u^t = \sum_R c_R K_R \sum_x x_{s_R} p_x^t = \sum_R c_R K_R E_{s_R}^t = \sum_{R:u\in P_R} c_R E_{s_R}^t - E_u^t \sum_{R: u=s_R}c_R,$$
where $P_R$ are the products of reaction $R$.
The inflow of molecules in place $u$ is proportional to the average numbers of the parent molecules of $u$.
Their proportionality constants are equal to reaction intensities.
Outflow using any reaction is permitted, possibly leading to the \textit{cemetery}, e.g. in case of the second fragmentation. This technical assumption is also needed to correctly solve the resulting ODEs, described later on.

The last formula can be rewritten as $ \dot E_u^t = \sum_{v\rightarrow u} \lambda_{vu} E_v^t - \lambda_{uu}E_u^t$ to underline the dependence of the average amount of $u$ upon respective average levels of species it originates from, $v$, that we call $u$'s parents.
The presented system of ODEs is recursive and can be solved from the root $r$ downwards.
For the root, $\dot E_r^t=-\lambda_{rr} E_r^t$, so that function $E_r^t= Ne^{-\lambda_{rr}t}$ solves it.
If we know solution of all ancestors of $u$ we can explicitly solve the problem for $u$. To see this, consider first a child $u$ of the root species $r$.
Then, $\dot E_u^t= E_r^t(t)-\lambda_{uu} E_u^t$ which results in $E_u^t= \frac{N}{\lambda_{rr}-\lambda_{uu}}(e{-\lambda_{uu}t}-e^{-\lambda_{rr}t})$ because
$E_u^0=0$, provided that $\lambda_{rr} \not= \lambda_{uu}$. In general, one has to consider a linear combination of solution of all ancestors of $u$ rather then only the root solution and solve $E_u^t=e^{-\lambda_{uu}t} \int_0^t e^{\lambda_{uu}s} f_{>u}(s)ds$.
Then, one can prove that the general solution is $E_u^t=\sum_{v>u}b_{vu}e{-\lambda_{vv}t}-b_{uu}e^{-\lambda_{uu}t}$, where $b_{vu}= \frac{1}{\lambda_{uu}-\lambda_{vv}} \sum_{w: v\geq w \rightarrow u} \lambda_{wu}b_{vw}$ and $b_{uu}=-\sum_v b_{vu}$.
Also this result relies on $\lambda_{vv} \not= \lambda_{uu}$ for all parents $v$ of $u$.
% Thus, $\dot E_u^t=f_{>u}(t)-\lambda_{uu} E_u^t$. Given that $E_u^0=0$ for all $u_r$, the solution to this equation is, by straightforward integration, $E_u^t=e^{-\lambda_{uu}t} \int_0^t e^{\lambda_{uu}s} f_{>u}(s)ds$.
% Consider that $u$ is one of the sons of root $r$: plug in $f_{>u}(s)=E_r^s$ above and perform integration to obtain $E_u^t= \frac{N}{\lambda_{rr}-\lambda_{uu}}(e{-\lambda_{uu}t}-e^{-\lambda_{rr}t})$.
% This holds only when $\lambda_{vv}\not=\lambda_{uu}$, which is true because of the assumptions on the form of intensities.
%
% Then, one can prove that the general solution is $E_u^t=\sum_{v>u}b_{vu}e{-\lambda_{vv}t}-b_{uu}e^{-\lambda_{uu}t}$, where $b_{vu}= \frac{1}{\lambda_{uu}-\lambda_{vv}} \sum_{w: v\geq w \rightarrow u} \lambda_{wu}b_{vw}$ and  $b_{uu}=-\sum_v b_{vu}.$

The required inequalities hold when one assumes that $\lambda_u = I q_i^2 P_{R_u}$, where $I$ is the overall intensity of all reactions, $q_u$ is the charge of molecules in place $u$, and $P_{R_u}$ is the probability of $R_u$ given that some reaction occurred. Then $\lambda_{vv} - \lambda_{uu} = I \sum_{R} P_R (q_v^2  - q_u^2) > 0.$
The last inequality holds because $u$ has a lower charge state than any of its ancestor
(as all reactions reduce the charge state by at least one).
The quadratic dependence on charge, $q_u^2$, is not merely technical: it can be motivated by theoretical considerations \cite{McLuckey1999-su}.

For ETnoD and PTR respectively $P_{R_u}=P_\text{PTR}$ and $P_{R_u}=P_\text{ETnoD}$.
The case of ETD is more complex, as it can cleave different bonds. We denote the probability of the cleavage of the $l^\text{th}$ bond by $P_{\text{ETD}_l}$. Additionally, one has to divide the $q-1$ protons and $g$ quenched protons between both the \ce{c} and \ce{z} fragments, so that $q_c+q_z=q-1$ and $g_c+g_z = g$. In its turn, the division of remaining $q-1$ protons and $g$ quenched protons depends on the available number of basic amino-acids on both fragments, denoted by $B_c$ and $B_z$. It is assumed to occur with probability $P_l(q_c, g_c) = \frac{ \binom{B_c}{q_c}\binom{B_z}{q_z} }{ \binom{B_c+B_z}{q-1} } \frac{ \binom{B_c - q_c}{g_c} \binom{B_z-q_z}{g_z} }{ \binom{B_c+B_z-q+1}{g} }$,
which corresponds to placing the $q_c$ charges on the basic sites of the fragment peptide followed by placing $g_c$ quenched protons on the remaining free basic sites. Placing quenched protons before charges would result in the same formula.

To conclude, the probability of the ETD on the $l^\text{th}$ amino acid with a given proton distribution among fragments equals $P_{R_u}=P_{\text{ETD}_l} P_l(q_c, g_c)$. The probability of observing any ETD reaction, $P_\text{ETD}$, can be obtained by summing $P_{\text{ETD}_l}$ over possible cleavage sites.

Overall, the model is parametrized by the total reaction intensity $I$, two reaction probabilities $(P_\text{PTR}, P_\text{ETnoD})$ and $L$ ETD reaction probabilities $P_\text{ETD}$. For a given vector of model parameters  we can obtain the average number of cations for each molecular species at time $t=1$, $e_u^1 =e_{u\theta}$. On the other hand, using {\sc MassTodon} we obtain the estimates $y_u$ of the percentual content of that molecular species in the experimental mass spectrum. For a given $\theta$, we measure the error using a logarithm of a sum of squared errors, $$d(e_\theta,y) = \log \sum_u (e_{u\theta}-y_u)^2.$$
The element $\theta^*$ that minimizes the error is our estimate of the true value of the parameters.


\section{Validation \& Results}

Numerical simulations of ETD reaction were performed to assess the quality of the fitting procedure in fully controlled conditions. The simulation was performed as follows: we start with a given number $N$ of substance P precursor cations. We simulate the electrospray ionization by placing a given number of protons on randomly chosen basic amino acids. Then, we simulate the Markov Jump Process using standard simulation techniques \cite{Gillespie1977-fr}, noting that our process can be simulated as if the cations reacted independently of each other. Ions that find themselves in the same state at the end of the simulation are aggregated. The resulting counts of ions simulate results obtainable with the {\sc MassTodon}.

We also analyze the robustness of the fitting procedure to noisy or missing data. The random noise is modelled by adding gaussian noise to the counts, with zero mean and standard deviation expressed as a given percentage of the count. Missing data is modelled by randomly removing a given proportion of the peaks. Finally, the counts obtained in this way are normalized to one. Altogether, the simulation was repeated $100$ times for $20-1$ different values of data distortion parameters (one pair of parameters was repeated), see Fig.\ref{fig::kokos}.

We use experimental data for Substance P. The precise experimental setting is described in detail in \cite{Lermyte2015-eb}.
\begin{figure}[h]
        \center
        \includegraphics[width=.9\textwidth]{kokos.png}
        \caption{ Relative errors of the fitting procedure on in silico Substance P data. The known true values of parameters are respectively $P_\text{ETD}=30\%, P_\text{ETnoD}= 25\%, P_{PTR}= 45\%$. Cleavage probabilities were assumed to be uniform. Each box-plot summarizes the results of $100$ independent simulations: whiskers denote the first and ninth decile and the box lids - the first and third quartiles. The left panel presents the response of the relative error of the estimates to the increasing amount of noise in the intensities reported by {\sc MassTodon}. In that case, we did not additionally model missing data (by molecular species). On the right panel, we study the impact of random removal of information on the molecular species, both in noiseless conditions (in gold) and with a modest amount of noise (standard deviation set to $20\%$ of the intensity of the simulated molecule).
        }\label{fig::kokos}
\end{figure}
% \section{Results \& Discussion}
We have tested the model and the fitting procedure on both simulated and experimental data to assess their robustness and to estimate real reaction intensities.

\textbf{In silico.} The model has been fitted to in silico data obtained as described above. The fitting procedure turned out to be fairly robust against moderate noise (up to $ \approx 50\%$ of the signal intensity) and missing data, see Fig.~\ref{fig::kokos}. The results of the fitting procedure are unbiased. On noiseless data and data with moderate amount of noise (up to $50\%$ of variation in simulated intensities), the model was able to predict the reaction intensities with very high accuracy (only after introducing more than $25\%$ of peak variation do the estimates start to surpass the limit of $50\%$ relative error in more than 20 percent of cases). Generally, the obtained estimates were unbiased, with reasonably low variance.

\textbf{Fitting to experimental data.} The model has been fitted to $53$ Substance P spectra, obtained at various travelling-wave height/velocity combinations as described in the Materials and Methods section. After fitting the model to the data, the validity of the model was further investigated by computing the percentage of the experimental spectrum explained by the theoretically predicted spectrum. We call this value the \textit{Explanation Percentage} (EP) and define it to be the common part of a theoretical and experimental spectrum. Since both spectra are normalized so that they sum to one, the Explanation Percentage can be expressed in a simple formula,
$ EP = \sum_u \min\{y_u, e_u^\text{norm}\}.$
Note, that because of normalization, $0 \leq EP \leq 1$. We present the Explanation Percentage calculated for considered data sets in Fig.~\ref{fig::melon} (two panels in the upper left corner): the typical values are between $40\%$ and $98\%$, mostly around $80\%$. These results are very promising given that the assumption that process intensities are constant, is rather strict, as discussed later on.

\begin{figure}[h]
        \center
        \includegraphics[width=1.1\textwidth]{melon.png}
        \caption{ Results of fitting to experimental data preprocessed by the {\sc MassTodon} software. Two black-and-white plots in the top left corner report the Explanation Percentages calculated for the input data. Below, we present results of fitting the model to the fragments obtained at Wave Height $= 1.5$ and Wave Velocity $= 2250$. Intensities of the \ce{c} fragments are plotted to the left of the black division line and are on the same height as the intensities of their corresponding \ce{z} fragments. The figure aggregates results for different charges and quenched charges. The right side of the panel presents estimates of intensities (top and bottom) and estimates of reaction probabilities (middle).
        }\label{fig::melon}
\end{figure}
The predicted total intensity of all reactions, I, can be found between $10^{-3}$ and $10$, see Fig.~\ref{fig::melon}. In regions of high reaction intensity (wave height between $0$ and $0.3$) the spectra are much more informative and even then the model can explain around $70\%$ of the input information. Similar results are obtained for different values of wave velocity. however, in the regions of high intensity (wave velocity above $1750$) the model explains around $75\%$ of the input.

In the bottom left corner of Fig.~\ref{fig::melon} we present the comparison of peak heights obtained with {\sc MassTodon} and those fitted by described procedure. The fragments have been aggregated over protons and quenched protons to simplify the plot. Note that in the input data there are more (N-terminal) \ce{c} fragments than (C-terminal) \ce{z} fragments, with the exception of the \ce{z11} fragment, corresponding to the loss of \ce{NH3} from the N terminus. This lack of symmetry is stronger than that expected in the current setting, but likely related to the asymmetric distribution of basic sites within the substance P sequence: the N-terminal amino group, the arginine residue, and the lysine residue are all located on the N-terminal side of the peptide.

\section{Discussion \& Conclusions}
We present a kinetic model of electron transfer driven reactions. The obtained results are promising for future work, as usually the model can explained around $80\%$ of the observed intensities of all molecular species. The model is based on stochastic foundations and so the estimated parameters have a probabilistic interpretation, such as the probability of a given cleavage or reaction.

Due to its simplicity, the model described here can be used in further fundamental research into the ETD mechanism, as a discrepancy between experimental observations and the model predictions is expected to have a relatively straightforward physical interpretation. For instance, the underestimation of the asymmetry of corresponding \ce{c} and \ce{z} fragment intensity in the current results might indicate that a more sophisticated model of protonation sites should be used (e.g. one that accounts for electrostatic repulsion, see \cite{Morrison2016-wc}) Similarly, using the {\sc MassTodon} software, it has been recently shown \cite{Lermyte2017-zt} that the observed ratio of PTR to ETnoD depends on protein conformation for intermediate charge states of ubiquitin. A more detailed  analysis could be easily performed (and similar dependencies thus revealed) using ETDetective.

% The main limitation of the presented methodology is currently the precursor charge state, as the size of the reaction graph grows polynomially  with the number of reactions and exponentially with the precursor charge. This restriction has been met in the literature in the context of model considered by Zhang, who had to simplify the basic model developed for singly and doubly charged precursor, in order to extend it to triply and quadruply charged precursor. An example possibility to overcome this problem is by using Monte Carlo approach, like the Approximate Bayesian Computing.

As mentioned in the Introduction, our model, being a kinetic model, is somewhat similar in nature to that presented by Zhang \cite{Zhang2010-fp}. However, there are many differences in the conceptual approach to the problem. The earlier model is derived from first principles of statistical physics, whereas that proposed by us is much more phenomenological: the physical content is reflected only in the construction of possible states and enumeration of ways of how one state can be modified into another state. Knowing this, we cast the problem into the well studied setting of continuous time Markov Jump Processes. Because of that, the number of parameters that describes our model is fairly limited, in contrast to the approach described in  \cite{Zhang2010-fp}. Another difference is that we do not estimate any parameters common to more than one data set: we can estimate our model completely based on one spectrum alone. This allows us to compare reaction intensities for different experimental conditions. As shown in the Results section, this lets us to simplify the comparison of many mass spectra gathered in different experimental conditions and summarize the results in a convenient way using just a few dimensions. Note that in precisely the same way one can compare spectra coming from different instruments, which is important to properly design the experiment, see \cite{Lermyte2015-lm}.

A natural way for this article to proceed is to explain the influence of the instrumental settings and experimental conditions on the reaction intensity and cleavage preferences. This can be investigated using statistical methodology, like the generalized linear models, Dirichlet regression in particular.

        % What is more, estimating the reaction intensities does not prevent us from estimating the deep parameters of the phenomenon, such as the activation energy in Arrhenius equations or the parameters describing the effective temperature. The task is even greatly simplified, as one can apply any traditional statistical and/or machine learning techniques used to model the the functional dependence of the intensities estimated from individual spectra given the other descriptors that may include specific experimental settings of the instrument. For instance, the analysis of the dependence of conditional probabilities PR on the deep parameters could be performed using Dirichlet regression or sparse Dirichlet regression \cite{Chen2013-zb}. Finding and fine-tuning the optimal tools for the further analysis of the data obtained by the methods outlined in this article is the natural way for this research to proceed.

{\textbf{Acknowledgements.}
        This work was partially supported by the National Science Centre grants number 2013/09/B/ST6/01575, 2014/12/W/ST5/00592 and 2015/17/N/ST6/03565, the SBO grant \textit{InSPECtor} (120025) of the Flemish agency for Innovation by Science and Technology (IWT). The authors thank the Research Foundation – Flanders (FWO) for funding a Ph.D. fellowship (F.L.). The Synapt G2 mass spectrometer is funded by a grant from the Hercules Foundation – Flanders.}
\bibliographystyle{splncs03.bst}
% {\footnotesize\bibliography{ISBRA_pubs}}
{\tiny\bibliography{paperpile}}

\end{document}
